### PR TITLE
Additional cmake cleanup

### PR DESCRIPTION
**Pull request by larshg**
Fixes the remaining issues in #2649
With the last commit, #2551 should be fixed too.

### CODE REVIEW

File: cmake/pcl_targets.cmake [hunk 1]
```diff
@@ -110,8 +110,10 @@ macro(PCL_SUBSYS_DEPEND _var)
   if(ARGS_OPT_DEPS)
     SET_IN_GLOBAL_MAP(PCL_SUBSYS_OPT_DEPS ${_name} "${ARGS_OPT_DEPS}")
   endif()
+
   GET_IN_MAP(subsys_status PCL_SUBSYS_HYPERSTATUS ${_name})
-  if(${_var} AND (NOT ("${subsys_status}" STREQUAL "AUTO_OFF")))
```
Comment (mvieth) [inline cmake/pcl_targets.cmake:114]: Isn't this set by `PCL_SUBSYS_OPTION`? So it should exist?
As I understand this check, if build is `FALSE`, it is unnecessary to check the dependencies

Comment (larshg) [reply cmake/pcl_targets.cmake]: Must have written wrong when I tested it. It exist indeed.

File: apps/modeler/CMakeLists.txt [hunk 2]
```diff
@@ -6,19 +6,13 @@ set(REASON "")
...
 # QVTK?
 if(NOT HAVE_QVTK)
-  set(DEFAULT AUTO_OFF)
+  set(DEFAULT OFF)
   set(REASON "VTK was not built with Qt support.")
-elseif(NOT ${DEFAULT} STREQUAL "AUTO_OFF")
-  set(DEFAULT TRUE)
-  set(REASON)
+else()
+  set(DEFAULT ON)
```
Comment (mvieth) [inline apps/modeler/CMakeLists.txt:12]: This would change the behaviour, right? Previously, the default was always FALSE

Comment (larshg) [reply apps/modeler/CMakeLists.txt]: Changed back to default off.

**Review by mvieth — APPROVED** [review]

**Pull request merged**